lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -217,7 +217,7 @@ synchronized void transferQueuedPresenceMessages(List<QueuedMessage> messagesToT
 
     private boolean attachResume;
 
-    private void attachImpl(final boolean forceReattach, final CompletionListener listener) throws AblyException {
+    private void attachImpl(final boolean forceReattach, final CompletionListener listener, ErrorInfo msgErr) throws AblyException {
```

Comment:
I suggest using `reattachmentReason` instead of `msgErr`

Reply:
or `attachmentReason` because `attachImpl` is triggered by both `explicit` and `implicit` attach

Reply:
or simply `reason`, since `attachImpl` method already implies it's channel attach operation wdyt?

Reply:
`reattachmentReason` a bit more specific, cause this is not null only in the case of reattachment, but I have no objections to the `reason`

Reply:
Okay, updated to `reattachmentReason`.
Not sure why `should_have_serial_action_createdAt` is failing, it's also failing on `main`. 
Probably sandbox issue ?